### ISSUE TITLE
gdb: "Reply contains invalid hex digit 0 " after hitting hardbreak

### ISSUE THREAD

**Issue by snickerbockers** [washingtondc-emu/washingtondc#15]
I'm not sure what's going on but it looks like this was introduced by 8ea736a2723283623e2a9c79173646941f4ddd2b .

```
$ gdb-multiarch 320x240.elf                                                                                                                                                                                                                   
GNU gdb (Ubuntu 7.11-0ubuntu1) 7.11                                                                                                                                                                                                           
Copyright (C) 2016 Free Software Foundation, Inc.                                                                                                                                                                                             
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>                                                                                                                                                                 
This is free software: you are free to change and redistribute it.                                                                                                                                                                            
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"                                                                                                                                                                    
and "show warranty" for details.                                                                                                                                                                                                              
This GDB was configured as "x86_64-linux-gnu".                                                                                                                                                                                                
Type "show configuration" for configuration details.                                                                                                                                                                                          
For bug reporting instructions, please see:                                                                                                                                                                                                   
<http://www.gnu.org/software/gdb/bugs/>.                                                                                                                                                                                                      
Find the GDB manual and other documentation resources online at:                                                                                                                                                                              
<http://www.gnu.org/software/gdb/documentation/>.                                                                                                                                                                                             
For help, type "help".                                                                                                                                                                                                                        
Type "apropos word" to search for commands related to "word"...                                                                                                                                                                               
Reading symbols from 320x240.elf...done.                                                                                                                                                                                                      
(gdb) target remote localhost:1999                                                                                                                                                                                                            
Remote debugging using localhost:1999                                                                                                                                                                                                         
0x8c010000 in start ()                                                                                                                                                                                                                        
(gdb) hbreak timer_ms_handler                                                                                                                                                                                                                 
Hardware assisted breakpoint 1 at 0x8c0318ea: file timer.c, line 150.                                                                                                                                                                         
(gdb) c                                                                                                                                                                                                                                       
Continuing.                                                                                                                                                                                                                                   
Reply contains invalid hex digit 0                                                                                                                                                                                                            
(gdb) q                                                                                                                                                                                                                                       
A debugging session is active.                                                                                                                                                                                                                
                                                                                                                                                                                                                                              
        Inferior 1 [Remote target] will be killed.                                                                                                                                                                                            
                                                                                                                                                                                                                                              
Quit anyway? (y or n) n                                                                                                                                                                                                                       
Not confirmed.                                                                                                                                                                                                                                
(gdb) bt                                                                                                                                                                                                                                      
/build/gdb-sqtpXm/gdb-7.11/gdb/target.c:1398: internal-error: target_xfer_partial: Assertion `*xfered_len > 0' failed.                                                                                                                        
A problem internal to GDB has been detected,                                                                                                                                                                                                  
further debugging may prove unreliable.                                                                                                                                                                                                       
Quit this debugging session? (y or n) y                                                                                                                                                                                                       
                                                                                                                                                                                                                                              
This is a bug, please report it.  For instructions, see:                                                                                                                                                                                      
<http://www.gnu.org/software/gdb/bugs/>.                                                                                                                                                                                                      
                                                                                                                                                                                                                                              
/build/gdb-sqtpXm/gdb-7.11/gdb/target.c:1398: internal-error: target_xfer_partial: Assertion `*xfered_len > 0' failed.                                                                                                                        
A problem internal to GDB has been detected,                                                                                                                                                                                                  
further debugging may prove unreliable.                                                                                                                                                                                                       
Create a core file of GDB? (y or n) n                                                                                                                                                                                                         
```

**Comment by snickerbockers** [comment]
Possibly due to some sort of socket-buffering by libevent.  Maybe also related to issue #6.

**Comment by snickerbockers** [comment]
resolved as of commit 99e24cf1bf3c46d6981d51909876cc50d50c8b65
